### PR TITLE
Resign the policy tab

### PR DESCRIPTION
**Pull request by Ankit-Keshari-Vituity**
"Redesign the Policies tab

* Change the complete UI of policies Tab
* Added a paragraph under Manage Policies --> Manage access for users & groups of DataHub using Policies
* Add a Table instead of a list to show the data
* Add columns Name, Type, Description,  Actors, State, and Actions
* Table Columns:
- Name --> Clicking on name opens the details modal
- Type --> "Platform" or "Metadata"
- Description
- Actors
- State -> Active vs Inactive
- Actions: Edit Policy, Deactivate / Activate Policy, Delete Policy

policy

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by Ankit-Keshari-Vituity** [comment]
Fixed the CSS issue.
- Border fix for state column
- Tags for "All Users" and "All Groups" on the right side of AvatarGroup (if fields allUsers / allGroups are true)

policies_css

File: datahub-web-react/src/app/policy/PoliciesPage.tsx [hunk 1]
```diff
@@ -239,10 +453,8 @@ export const PoliciesPage = () => {
                 <PolicyDetailsModal
                     policy={focusPolicy}
                     visible={showViewPolicyModal}
-                    onEdit={onEditPolicy}
                     onClose={onCancelViewPolicy}
-                    onRemove={onRemovePolicy}
-                    onToggleActive={onToggleActive}
+                    privileges={privileges}
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/policy/PoliciesPage.tsx:457]: what if we had a prop called 

"getPrivilegeDisplayName(type: string) => string"

File: datahub-web-react/src/app/policy/PolicyDetailsModal.tsx [hunk 2]
```diff
@@ -1,20 +1,17 @@
...
 
 type Props = {
-    policy: Omit<Policy, 'urn'>;
+    policy: any;
     visible: boolean;
-    onEdit: () => void;
     onClose: () => void;
-    onRemove: () => void;
-    onToggleActive: (value: boolean) => void;
+    privileges: any;
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/policy/PolicyDetailsModal.tsx:14]: Change this type to Array<String>

File: datahub-web-react/src/app/policy/PolicyDetailsModal.tsx [hunk 3]
```diff
@@ -37,133 +34,142 @@ const ThinDivider = styled(Divider)`
...
+                            {policy?.resolvedGroups?.map((group) => (
+                                <CustomAvatar
+                                    size={28}
+                                    name={group?.name}
+                                    url={`/${entityRegistry.getPathName(EntityType.CorpGroup)}/${group.urn}`}
+                                    isGroup
+                                />
+                            ))}
+                        </Col>
+                        <Col flex={1} style={{ display: 'flex', justifyContent: 'end' }}>
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/policy/PolicyDetailsModal.tsx:169]: Try to use styled components any time there is inline styling overrides

File: datahub-web-react/src/app/policy/PoliciesPage.tsx [hunk 4]
```diff
@@ -197,35 +253,193 @@ export const PoliciesPage = () => {
...
+        },
+        {
+            title: '',
+            dataIndex: '',
+            key: 'x',
+            render: (_, record: any) => (
+                <ActionButtonContainer>
+                    <Button
+                        disabled={!record?.editable}
+                        style={{ marginRight: 16 }}
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/policy/PoliciesPage.tsx:349]: Move to styled components

File: datahub-web-react/src/app/policy/PoliciesPage.tsx [hunk 5]
```diff
@@ -197,35 +253,193 @@ export const PoliciesPage = () => {
...
+                                        url={`/${entityRegistry.getPathName(EntityType.CorpUser)}/${user.urn}`}
+                                        photoUrl={
+                                            user?.editableProperties?.pictureLink ||
+                                            user?.editableInfo?.pictureLink ||
+                                            undefined
+                                        }
+                                    />
+                                ))}
+                            </Col>
+                            <Col flex={1} style={{ display: 'flex', justifyContent: 'end' }}>
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/policy/PoliciesPage.tsx:309]: Remove inline styling

File: datahub-web-react/src/app/policy/PoliciesPage.tsx (hunk 1 above)
Comment (jjoyce0510) [inline datahub-web-react/src/app/policy/PoliciesPage.tsx:457]: privileges={getPrivileges(focusPolicy)}

Comment (jjoyce0510) [reply datahub-web-react/src/app/policy/PoliciesPage.tsx:457]: And remove the state

File: datahub-web-react/src/app/policy/PoliciesPage.tsx [hunk 6]
```diff
@@ -139,10 +173,31 @@ export const PoliciesPage = () => {
         setShowPolicyBuilderModal(false);
     };
 
+    const getPrivileges = (policy: Policy) => {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/policy/PoliciesPage.tsx:176]: rename to "getPrivilegeNames"

File: datahub-web-react/src/app/policy/PoliciesPage.tsx [hunk 7]
```diff
@@ -197,35 +253,193 @@ export const PoliciesPage = () => {
...
+                        <DeleteOutlined />
+                    </Button>
+                </ActionButtonContainer>
+            ),
+        },
+    ];
+
+    const tableData = policies?.map((policy) => ({
+        urn: policy?.urn,
+        policy,
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/policy/PoliciesPage.tsx:387]: Do we need this here? Or can we remove

File: datahub-web-react/src/app/policy/PoliciesPage.tsx [hunk 8]
```diff
@@ -197,35 +253,193 @@ export const PoliciesPage = () => {
...
-                        total={totalPolicies}
-                        showLessItems
-                        onChange={onChangePage}
-                        showSizeChanger={false}
-                    />
-                </PaginationContainer>
+                <PoliciesHeaderContainer>
+                    <PoliciesTitle level={2}>Manage Policies</PoliciesTitle>
+                    <Typography.Paragraph type="secondary">
+                        Manage access for users & groups of DataHub using Policies.
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/policy/PoliciesPage.tsx:410]: "Manage access for Users & Groups of DataHub using Policies."

File: datahub-web-react/src/app/policy/PoliciesPage.tsx (hunk 1 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/policy/PoliciesPage.tsx]: Changed and committed

File: datahub-web-react/src/app/policy/PoliciesPage.tsx (hunk 8 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/policy/PoliciesPage.tsx]: Changed and committed

**Review by shirshanka — APPROVED** [review]

**Review by shirshanka — APPROVED** [review]

**Pull request merged**